replica/table.cc

```diff
@@ -1010,11 +1003,13 @@ void table::update_stats_for_new_sstable(const sstables::shared_sstable& sst) no
 
 future<>
 table::do_add_sstable_and_update_cache(sstables::shared_sstable sst, sstables::offstrategy offstrategy, bool trigger_compaction) {
+    compaction_group& cg = compaction_group_for_sstable(sst);
```

Comment:
Yes, it's better this way.

As a follow up, we have to protect the tablet from migrating away at a higher level, probably at the tablet topology coordinator level, to serialize migration with repair, load-and-stream, or any other process that streams data into the tablet to prevent leaving behind any sstable not supposed to be on the migration-source node as it might cause data resurrection if the tablet is moved back to that node.

Cc @tgrabiec @asias